### ISSUE TITLE
When generating caption data using the coca model, there are many incomplete sentences or repetitive meaningless numbers in the results. Why is that?

### ISSUE THREAD

**Issue by phil-zd** [apple/ml-mobileclip-dr#4]
Code:              
```
    image_batch = torch.stack(image_tensors).to(device)
    generated = model.generate(
          image=image_batch,
          generation_type="top_p",
          top_p=0.9,
          fixed_output_length=True,
          max_seq_len=77,
          seq_len=77,
  )
```

results:
```
1. a 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
2. a close up of steven spielberg . the person wearing a white shirt is wearing long black hair . the camera is blurry , and the side of
3. a kitchen with dark cabinets . at the top of the frame , is a single light . the middle section is on each side of the frame
4. fahad 2 2 2 2 tarafından logo design for " rmf consulting " için no 2 2
```

**Comment by fartashf** [comment]
This behavior is expected of some models and particularly ones trained with longer context length. We recommend using the model fine-tuned on MSCOCO-38k with context length 77:
[mobileclip2_coca_dfn2b_s13b_mscoco38k_s12m_context77](https://huggingface.co/apple/mobileclip2_coca_dfn2b_s13b_mscoco38k_s12m_context77)
Please refer to Table 6 in the [paper](https://arxiv.org/abs/2508.20691) for the performance of alternatives.

We have further organized the models on HuggingFace in groups based on their context length:
https://huggingface.co/collections/apple/mobileclip2-68ac947dcb035c54bcd20c47

Additionally, CoCa-DFN2B model may be used as a base for fine-tuning on other datasets:
[mobileclip2_coca_dfn2b_s13b_context77](https://huggingface.co/apple/mobileclip2_coca_dfn2b_s13b_context77)

Note that we observed training on synthetic captions with higher chance of repetitions may still lead to high zero-shot classification/retrieval performance. Notably, DFN-2B-DOCCI-context256:
[mobileclip2_coca_dfn2b_s13b_docci_s12m_context256](https://huggingface.co/apple/mobileclip2_coca_dfn2b_s13b_docci_s12m_context256)

**Comment by phil-zd** [comment]
Thank you very much for your reply. I'm sorry that the information I provided was not clear enough.
In fact, I used 21 coca models open-sourced from [huggingface](https://huggingface.co/collections/apple/mobileclip2-68ac947dcb035c54bcd20c47) to generate captions for the same batch of image data, and the interfaces I used were all:
```
                    if image_tensors:
                        image_batch = torch.stack(image_tensors).to(device)
                        generated = model.generate(
                            image=image_batch,
                            generation_type="top_p",
                            top_p=0.9,
                            fixed_output_length=False,
                            max_seq_len=77,
                            # seq_len=77,
                        )

                        decoded_texts = [
                            open_clip.decode(g)
                            .split("<start_of_text>")[-1]
                            .split("<end_of_text>")[0]
                            .strip()
                            for g in generated
                        ]
```
However, most of the captions generated by these models have the problem of incomplete sentences. A small number of them will also have irrelevant repetitive characters. The proportion of problems varies among different models. I'm very confused about this. Thank you for your help

**Comment by fartashf** [comment]
Thanks for sharing your code. It is expected that captions generated with these models are not complete sentences and closer to short image captions. For example, testing [this coca model](https://huggingface.co/apple/mobileclip2_coca_dfn2b_s13b_mscoco38k_s12m_context77) on [this image](https://github.com/apple/ml-mobileclip/blob/main/docs/fig_accuracy_latency_v2.png), the following captions are expected:
```
a line graph comparing latency of image encoder scores
a line graph with ten different types of points
a scatter plot is explained in the diagram
a line graph with four horizontal bars over a sampling of text encoder sizes
a line graph that is showing is continual
a scatter plot is split off series in excel
a line graph is chart of dataset compartments
```
Not all of these examples are grammatically correct but they are good for training image-text models for zero-shot classification/retrieval.

Please note that for models trained with context length 256, the OpenCLIP model config needs to be modified to set `context_length=255`. The instructions or in [this repository](https://github.com/apple/ml-mobileclip). Also, we generally ignore all the text after the first period. So we use the following sample code:
```
syn_text = open_clip.decode(syn_text).split("<end_of_text>")[0].split("<start_of_text>")[-1].split(".")[0].rstrip()
```
and make sure the model is in eval mode for slightly higher quality samples.
